packages/create-apollo-client/src/createApolloLinks.js

```diff
@@ -0,0 +1,207 @@
+// @flow
+
+import * as R from 'ramda';
+import { ApolloLink } from 'apollo-link';
+import { BatchHttpLink } from 'apollo-link-batch-http';
+import { onError } from 'apollo-link-error';
+import { FileUploadLink, TokenRefreshLink } from '@8base/sdk';
+
+
+/**
+ * Interface of the user auth info. Needs for the auth links correctly working.
+ */
+interface AuthState {
+  email: string,
+  organizationId: string,
+  accountId: string,
+  idToken: string,
+  refreshToken: string,
+}
+
+/**
+ * Interface of the Graphql error which arise on the server side.
+ */
+interface GraphQLError {
+  code: string,
+  details: Object,
+  message: string,
+}
+
+type GetErrorLinkParams = {
+  onGraphQLErrors?: (GraphQLError[]) => void,
+  onNetworkError?: (error: Object) => void,
+}
+
+type GetAuthStateParams = {
+  getAuthState?: () => AuthState,
+}
+
+type GetTokenRefreshLinkParams = {
+  onIdTokenExpired?: () => void,
+  onUpdateTokenFail?: ({ refreshToken: string, idToken: string }) => void,
+  onUpdateTokenSuccess?: (error?:Object) => void,
+}
+
+/**
+ * Interface of the createApolloLinks options
+ */
+type CreateApolloLinkParams = {
+  uri: string,
+  getAuthState?: () => AuthState,
+
+  links?: {
+    tokenRefresh?: GetTokenRefreshLinkParams & {
+      enable?: boolean,
+    },
+    fileUpload?: {
+      enable?: boolean,
+    },
+    auth?: {
```

Comment:
@ej9x How we can use client without auth link? For what?

Reply:
I remember we talking about the possibility of using 8base with outer auth services.

Reply:
Now, we can't use other auth services and client without auth link is unusable. @vorobeez what are you think about it?

Reply:
Then we should remove enable param from the tokenRefresh link too.